fishing fixes and additions

Generallemon07 opened the pull request:
# About the pull request

This PR makes it so that you are able to fish in oceans and rivers, it also adds 3 new fish types two that are able to be gutted for fish meat as well as a bunch of fun little drops you can get from fishing like ore, beer, shoes, and other novelties.


# Explain why it's good for the game

It would be a good RP addition for fishing and having fun. It doesn't really throw off the balance at all as far as I looked, its just a fun camping thing.


# Testing Photographs and Procedure
Making sure fishing pole only went in water and not anywhere else: 
![image](https://github.com/user-attachments/assets/98cdcec8-b742-4afe-bc58-075788d0c45c)
Making sure the sprites and gutted item drops were good (catfish and gutted bluegill sprites were later changed): 
![image](https://github.com/user-attachments/assets/bdee67c1-edcc-48d9-8e3e-808dca0fe0ee)
Making sure the loot tables were being used while fishing (the clam placeholder was replaced with an adequate sprite and function):
![image](https://github.com/user-attachments/assets/e5f899a3-c4d0-4da7-bb04-86e9acf31d06)




Sprites of bass in a pile: 
![image](https://github.com/user-attachments/assets/8ae2a3c8-3a7a-43df-adc1-b480166a1c88)

After getting my sprite approved for use i added it as a new fish snacks item.



# Changelog
:cl:Sergeant_Lemon
add: new fish types to the old fishing code
add: added turf flag 'fishable_turf'
add: added items to fishing loot table
fix: fixed fishing rod code so it fishes on ocean and rivers
code: Changed code in the fishing category as well as meat, ore, and atoms
imageadd: sprites for new fish and pearl item
/:cl:

maxo1243 commented:
this is really cool, hoping this could be on pve too!

Generallemon07 commented:
> this is really cool, hoping this could be on pve too!

Thanks, this is my first PR, I tested it out and I think it will be a real treat to people who just wanna do something else than the regular gameplay loop, more RP elements too.

Blundir reviewed (changes requested):
tweak meat sprites, discussed in discord, also you said you'd improve bone sprites, so do that too while you are on it

Generallemon07 commented:
Changed base fillet sprite and the other sprites requested.

Git-Nivrak reviewed (changes requested):
As above

Generallemon07 commented:
Used initialize reagent container reagents to make fish no longer claim "nothing left" when eaten. Added four new sound effects for fishing pole and replaced placeholder sounds as requested. 
@Git-Nivrak

Frogge6645 commented:
Bravo activities.
